packages/big-design/src/components/StatelessPagination/spec.tsx

```diff
@@ -0,0 +1,184 @@
+import { render, screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import React from 'react';
+
+import 'jest-styled-components';
+
+import { StatelessPagination } from './index';
+
+test('render Stateless pagination component', async () => {
+  const { findByRole } = render(
+    <StatelessPagination
+      itemsPerPage={3}
+      itemsPerPageOptions={[2, 3, 5]}
+      onItemsPerPageChange={jest.fn()}
+      onNext={jest.fn()}
+      onPrevious={jest.fn()}
+    />,
+  );
+
+  const pagination = await findByRole('navigation');
+
+  expect(pagination).toMatchSnapshot();
+});
+
+test('render Stateless pagination component while overriding button labels', async () => {
+  render(
+    <StatelessPagination
+      itemsPerPage={3}
+      itemsPerPageOptions={[2, 3, 5]}
+      label="[Custom] Pagination"
+      localization={{ previousPage: 'Pagina previa', nextPage: 'Pagina siguiente' }}
+      onItemsPerPageChange={jest.fn()}
+      onNext={jest.fn()}
+      onPrevious={jest.fn()}
+      rangeLabel="[Custom range label]"
+    />,
+  );
+
+  const pagination = await screen.findByRole('navigation', { name: '[Custom] Pagination' });
+  const dropdown = await screen.findByRole('button', { name: '[Custom range label]' });
+  const prevPage = await screen.findByRole('button', { name: 'Pagina previa' });
+  const nextPage = await screen.findByRole('button', { name: 'Pagina siguiente' });
+
+  expect(pagination).toBeInTheDocument();
+  expect(dropdown).toBeInTheDocument();
+  expect(prevPage).toBeInTheDocument();
+  expect(nextPage).toBeInTheDocument();
+});
+
+test('trigger range change', async () => {
+  const changeRange = jest.fn();
+  const { findByText } = render(
+    <StatelessPagination
+      itemsPerPage={2}
+      itemsPerPageOptions={[2, 3, 5]}
+      onItemsPerPageChange={changeRange}
+      onNext={jest.fn()}
+      onPrevious={jest.fn()}
+    />,
+  );
+
+  const option = await findByText('Show 2 items');
+
+  await userEvent.click(option);
+  await userEvent.keyboard('{ArrowDown}{Enter}');
+
+  expect(changeRange).toHaveBeenCalled();
+  expect(option).toBeInTheDocument();
+});
```

Comment:
Not quite sure on the exact action being tested. Is this testing for selecting `2` in the dropdown or selecting `3`? 

Not sure if the initial click puts the current user selection at the parent element or at the first child element in the dropdown.

Reply:
that is what's happening, by default the currently highlighted option will be focussed on first interaction. Pressing down goes to the next one (3) in the list

the `Dropdown` component (which this uses under the hood) already has tests proving keyboard usability of this component: https://github.com/bigcommerce/big-design/blob/main/packages/big-design/src/components/Dropdown/spec.tsx#L326-L342

so I've updated this _test_ to use the more explicit form of selecting and clicking on items: https://github.com/bigcommerce/big-design/pull/1598/commits/8884d2bf38ebd818ee2fd7d13e56f7741d23bee8

Reply:
Yep, thanks kindly. 

The clicking is definitely more explicit, but the most helpful bit here is the `toHaveBeenCalledWith(3)`. If we went back to the keyboard interaction, you'd be much more easily able to puzzle out that if the second option is being tested for, the selection must start at the first option.